universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx

```diff
@@ -24,11 +24,13 @@ type TopUpWithFiatModal = 'none' | 'wait' | TopUpProvider;
 export const TopUpWithFiat = ({hideModal, setHeaderVisible, walletService, modalClassName, logoColor}: TopUpWithFiatProps) => {
   const [modal, setModal] = useState<TopUpWithFiatModal>('none');
   const [amount, setAmount] = useState('');
+  const [currency, setCurrency] = useState<string>('ETH');
   const contractAddress = walletService.getContractAddress();
   const relayerConfig = walletService.sdk.getRelayerConfig();
   const [paymentMethod, setPaymentMethod] = useState<TopUpProvider | undefined>(undefined);
 
-  const onPayClick = (provider: TopUpProvider) => {
+  const onPayClick = (provider: TopUpProvider, currency: string) => {
+    setCurrency(currency);
```

Comment:
I guess it should not be state. It can be just variable
I think this two line can run render 2 times
```
setCurrency(currency);
setModal(provider);
```